crates/api_models/src/subscription.rs

```diff
@@ -174,10 +192,16 @@ pub struct ConfirmSubscriptionRequest {
     pub customer_id: common_utils::id_type::CustomerId,
 
     /// Billing address for the subscription.
-    pub billing_address: Option<Address>,
+    pub billing: Option<Address>,
+
+    /// Shipping address for the subscription.
+    pub shipping: Option<Address>,
 
     /// Payment details for the invoice.
     pub payment_details: PaymentDetails,
```

Comment:
There are new fields added in `PaymentDetails` which shouldn't be override by the client, lets include fields which client can control